google/cloud/storage/client_sign_url_test.cc

```diff
@@ -107,6 +107,17 @@ TEST_F(CreateSignedUrlTest, V2SignRemote) {
   EXPECT_THAT(*actual, HasSubstr(expected_signed_blob_safe));
 }
 
+/// @test Verify that CreateV2SignedUrl() respects the custom endpoint.
+TEST_F(CreateSignedUrlTest, V2SignCustomEndpoint) {
+  std::string custom_endpoint = "https://storage.mydomain.com";
```

Comment:
nit: consider

```suggestion
  auto constexpr kCustomEndpoint = "https://storage.mydomain.com";
```

or

```cc
  auto const custom_endpoint = std::string{"http://storage.mydomain.com"};
```